src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -264,6 +335,18 @@ private Object readResolve() {
             storageAccountNameReferenceType = "new";
         }
         storageAccountName = getStorageAccountName(storageAccountNameReferenceType, newStorageAccountName, existingStorageAccountName);
+
+        if (StringUtils.isBlank(imageTopLevelType)) {
+            if (StringUtils.isNotBlank(image)
+                    || StringUtils.isNotBlank(imageOffer)
+                    || StringUtils.isNotBlank(imageSku)
+                    || StringUtils.isNotBlank(imagePublisher)) {
+                imageTopLevelType = "advanced";
```

Comment:
basic and advanced are used in multiple places, change them to string constants.